### PR TITLE
Avoid adding pip related messages if installation method is different

### PR DESCRIPTION
**Pull request by ajinkyau**
Fixes: #3541

### CODE REVIEW

File: src/ansiblelint/config.py [hunk 1]
```diff
@@ -204,7 +204,18 @@ def in_venv() -> bool:
 
 def guess_install_method() -> str:
     """Guess if pip upgrade command should be used."""
+    # pylint: disable=import-outside-toplevel
+    from importlib.metadata import distribution
```
Comment (ssbarnea) [inline src/ansiblelint/config.py:208]: Put it outside, no need to use this approach in this case. Only with ansible core we need these.

Comment (ajinkyau) [reply src/ansiblelint/config.py:208]: ack

File: src/ansiblelint/config.py [hunk 2]
```diff
@@ -152,7 +152,7 @@ class Options:  # pylint: disable=too-many-instance-attributes,too-few-public-me
     config_file: str | None = None
     generate_ignore: bool = False
     rulesdir: list[Path] = field(default_factory=list)
-    cache_dir_lock: FileLock | None = None
+    cache_dir_lock: FileLock | None = None  # type: ignore[valid-type]
```
Comment (ssbarnea) [inline src/ansiblelint/config.py:155]: I kinda doubt this is needed, probably filelock was not included as a dependency in mypy, which would be the correct way to address the issue.

**Review by ssbarnea — APPROVED** [review]

File: src/ansiblelint/config.py [hunk 3]
```diff
@@ -205,6 +205,14 @@ def in_venv() -> bool:
 def guess_install_method() -> str:
     """Guess if pip upgrade command should be used."""
     package_name = "ansible-lint"
+
+    try:
+        if distribution(package_name).read_text("INSTALLER").strip() != "pip":  # type: ignore[union-attr]
```
Comment (gotmax23) [inline src/ansiblelint/config.py:210]:
```
         if distribution(package_name).read_text("INSTALLER").strip() != "pip":  # type: ignore[union-attr]
```

This returns None if the file doesn't exist which can happen, as I don't think INSTALLER is a mandatory part of the dist-info directory, so calling .strip()` directly is incorrect. Mypy warns about this for a reason...

You can probably change this to

```
        if (distribution(package_name).read_text("INSTALLER") or "").strip() != "pip":
```

Comment (ajinkyau) [reply src/ansiblelint/config.py:210]: Updated in PR https://github.com/ansible/ansible-lint/pull/3553

**Pull request merged**